types/compilation/compilation.interfaces.ts

```diff
@@ -121,3 +122,28 @@ export type ToolResult = {
     stdout: ResultLine[];
     artifact?: Artifact;
 };
+
+export type CompilationInfo = {
+    mtime: any;
+    compiler: any;
```

Comment:
This compiler is base-compiler's `this.compiler`. That should at some point be partially represented by `types/compiler.interface.ts` when we're done at least passing every key in `lib/compiler-finder.js` to that interface file. Could probably set the type here if if not too many ts files complain (And extend it with Record<any, any> if they do)